### PR TITLE
Buttslap command

### PR DESCRIPTION
**Pull request by Tsugami**
A new command to you SLAP THAT THICC ASS of your friend (no homo). I also splitted CanvasTemplates in files in a folder bc i dont wanted to find the files that were uaing CanvasTemplates and i let it with the same name using inside index.js

### CODE REVIEW

**Comment by pedrofracassi** [comment]
English please

**Comment by Tsugami** [comment]
ready?

**Comment by Tsugami** [comment]
I have not installed gif-frames in package.json e-e'

File: src/commands/image-manipulation/buttslap.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+const { Command, CommandParameters, UserParameter } = CommandStructures;
+const { Attachment } = require('discord.js');
+
+class ButtSlap extends Command {
+    constructor(client) {
+        super(client);
+        this.aliases = ['slap'];
+        this.parameters = new CommandParameters(this, 
+            new UserParameter({ acceptBot: true, acceptSelf: false, missingError: 'commands:buttslap.noMentions' }),
+            new ({ acceptBot: true, acceptSelf: true, required: false })
```
Comment (davipatury) [inline src/commands/image-manipulation/buttslap.js]: ?

File: src/commands/image-manipulation/buttslap.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+            channel.startTyping();
+
+            let stream = await CanvasTemplates.ButtSlap(tom.displayAvatarURL, kettin.displayAvatarURL);
+            let attachment = new Attachment(stream, "image.gif");
+            
+            await channel.send({
+                content: t('commnads:buttslap.sucess', { _tom: tom, _kettin: kettin }),
+                files: [ attachment ]
+            });
```
Comment (davipatury) [inline src/commands/image-manipulation/buttslap.js]:
```js
channel.send(t('commnads:buttslap.success', { _tom: tom, _kettin: kettin }), attachment)```

File: src/commands/image-manipulation/buttslap.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+                tom = author;
+            };
+
+            channel.startTyping();
+
+            let stream = await CanvasTemplates.ButtSlap(tom.displayAvatarURL, kettin.displayAvatarURL);
+            let attachment = new Attachment(stream, "image.gif");
+            
+            await channel.send({
+                content: t('commnads:buttslap.sucess', { _tom: tom, _kettin: kettin }),
```
Comment (davipatury) [inline src/commands/image-manipulation/buttslap.js]: suc**c**ess

File: src/commands/image-manipulation/buttslap.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+            // ._.
+            if (!kettin) {
+                kettin = tom;
+                tom = author;
+            };
+
+            channel.startTyping();
+
+            let stream = await CanvasTemplates.ButtSlap(tom.displayAvatarURL, kettin.displayAvatarURL);
+            let attachment = new Attachment(stream, "image.gif");
```
Comment (davipatury) [inline src/commands/image-manipulation/buttslap.js]: `const` everywhere, use simple quotes too

File: src/index.js [hunk 5]
```diff
@@ -1,4 +1,4 @@
-const CommandStructures = require('./structures/command')
+const CommandStructures = require('./structures/command');
```
Comment (davipatury) [inline src/index.js]: no

File: src/locales/en-US/commands.json [hunk 6]
```diff
@@ -631,5 +631,11 @@
     "commandUsage": "<user to slap>",
     "noMention": "You need to mention someone",
     "success": "{{- _author}} slapped {{- slapped}} BAKA :<"
+  },
+
+  "buttslap": {
+    "commandUsage": "<user to slap>",
+    "noMention": "You need to mention someone",
+    "sucess": "{{- _tom}} hit the butt of {{- _kettin}}"
```
Comment (davipatury) [inline src/locales/en-US/commands.json]: suc**c**ess

File: src/utils/CanvasTemplates/ButtSlap.js [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+        
+        }
+
+        encoder.finish();
+
+        return FILE;
+    }
+}
+
+module.exports = ButtSlap;
```
Comment (davipatury) [inline src/utils/CanvasTemplates/ButtSlap.js]: 2 spaces indentation

File: src/utils/CanvasTemplates/NowPlaying.js [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+        // Modal style
+        ctx.fillStyle = '#FFFFFF'
+        ctx.globalCompositeOperation = 'destination-in'
+        ctx.roundRect(0, 0, WIDTH, HEIGHT, 10, true)
+    
+        return canvas.toBuffer()
+      }
+}
+
+module.exports = NowPlaying;
```
Comment (davipatury) [inline src/utils/CanvasTemplates/NowPlaying.js]: 2 spaces indentation

File: src/utils/CanvasTemplates/Profile.js [hunk 9]
```diff
@@ -0,0 +1,181 @@
...
+        // Modal
+        ctx.fillStyle = '#FFFFFF'
+        ctx.globalCompositeOperation = 'destination-in'
+        ctx.roundRect(CARD_X_MARGIN, CARD_Y_MARGIN, CARD_WIDTH, CARD_HEIGHT, 10, true)
+    
+        return canvas.toBuffer()
+      }
+}
+
+module.exports = Profile;
```
Comment (davipatury) [inline src/utils/CanvasTemplates/Profile.js]: 2 spaces indentation

File: src/utils/CanvasTemplates/Triggered.js [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+          encoder.addFrame(ctx)
+        }
+    
+        encoder.finish()
+    
+        return encoder.out.getData();
+      }
+}
+
+module.exports = Triggered;
```
Comment (davipatury) [inline src/utils/CanvasTemplates/Triggered.js]: 2 spaces indentation

File: src/utils/CanvasTemplates/index.js [hunk 11]
```diff
@@ -0,0 +1,8 @@
+module.exports = {
+    ButtSlap: require('./ButtSlap.js').render,
+    
+    tiggered: require("./Triggered.js").render,
```
Comment (davipatury) [inline src/utils/CanvasTemplates/index.js]: tiggered

File: src/utils/CanvasTemplates/index.js [hunk 12]
```diff
@@ -0,0 +1,8 @@
...
+    profile: require("./Profile.js").render,
+    nowPlaying: require("./NowPlaying.js").render,
+    leaderboard: require("./LeaderBoard.js").render
+}
```
Comment (davipatury) [inline src/utils/CanvasTemplates/index.js]: 2 spaces indentation

File: src/utils/Constants.js [hunk 13]
```diff
@@ -41,6 +41,7 @@ module.exports = {
 
   // Misc
   TRIGGERED_LABEL_PNG: 'src/assets/png/triggered_label.png',
+  BUTTSLAP_GIF = 'src/assets/gif/buffslap.gif',
```
Comment (davipatury) [inline src/utils/Constants.js]: `:`, not `=`

File: src/commands/image-manipulation/buttslap.js (hunk 2 above)
Comment (pedrofracassi) [reply src/commands/image-manipulation/buttslap.js]: `comnads`

File: src/index.js (hunk 5 above)
Comment (pedrofracassi) [reply src/index.js]: u

**Comment by pedrofracassi** [comment]
@Tsugami can you fix this stuff?

**Comment by perronosaurio** [comment]
fix the Conflicting files

**Comment by Doges** [comment]
> I fixed it, see if everything's okay now

not everything, solve the grammar thingy

**Comment by pedrofracassi** [comment]
`CanvasTemplates.js` is not used anymore, I believe. Templates now go in their own files.

**Comment by pedrofracassi** [comment]
Can we get this one working? I quite like it.

**Pull request closed without merging**